clang/test/Analysis/stream.c

```diff
@@ -376,3 +377,122 @@ void fflush_on_open_failed_stream(void) {
   }
   fclose(F);
 }
+
+void getline_null_file() {
+  char *buffer = NULL;
+  size_t n = 0;
+  getline(&buffer, &n, NULL); // expected-warning {{Stream pointer might be NULL}}
+}
+
+void getdelim_null_file() {
+  char *buffer = NULL;
+  size_t n = 0;
+  getdelim(&buffer, &n, '\n', NULL); // expected-warning {{Stream pointer might be NULL}}
+}
+
+void getline_no_return_check() {
+  FILE *file = fopen("file.txt", "r");
+  if (file == NULL) {
+    return;
+  }
+
+  char *line = NULL;
+  size_t len = 0;
+  getline(&line, &len, file);
+
+  if (line[0] == '\0') {} // expected-warning {{The left operand of '==' is a garbage value}}
+
+  free(line);
+  fclose(file);
+}
+
+void getline_after_eof() {
```

Comment:
I think `getline_after_eof`, `getline_feof`, `getline_feof_check` are checking not much more cases than `error_getline` (and `error_getdelim`) in _stream_error.c_ and can be removed. Even if not removed, such tests (that check `feof` and `ferror` and related warnings) should put into _stream_error.c_.

Reply:
Fair. I have removed them.